test/framework/src/test/java/org/elasticsearch/test/test/InternalTestClusterTests.java

```diff
@@ -255,6 +266,21 @@ public Settings transportClientSettings() {
         }
     }
 
+    private static List<Settings> addBootstrapConfiguration(Random random, List<Settings> allNodesSettings) {
+        final List<Settings> updatedSettings = new ArrayList<>(allNodesSettings);
+        while (true) {
```

Comment:
instead of the while loop, can you just filter all node settings that have `NODE_MASTER_SETTING.get(settings)` and then randomly select one of them? You're doing that filtering step anyway already further down below.

Reply:
Ok c793fc803bd8d4f6c666b04984454f7157e5211c. We need to filter the indices into the list, not the settings, because we need the index to replace the settings for the bootstrap node.